### DISCUSSION TITLE
Couldn't enable HTTPS.

### DISCUSSION THREAD

**Discussion by rahulr001** [jhuckaby/Cronicle#714] [Q&A]
## Summary

Need to create a custom domain in apache2 and in port 3013 which with ssl, and I update the config.json

	"server_comm_use_hostnames": true,
	"web_direct_connect": true,
	"web_socket_use_hostnames": true,
	
here I am able to access the landing page but couldn't connect to web socket, in web socket it should take hostname, but it is taking sever username and I could not figure out why it is happening could you please help me with this.

![Screenshot from 2024-01-25 12-12-42](https://github.com/jhuckaby/Cronicle/assets/109589036/03588e9c-c59c-4d52-a1b8-2bbab72d486c)

Thank you!

### Operating system and version?

Ubuntu 22

### Node.js version?

V22

### Cronicle software version?

Latest

### Are you using a multi-server setup, or just a single server?

Single server
### Are you using the filesystem as back-end storage, or S3/Couchbase?

Couchbase

**Comment by jhuckaby** [comment]
I'm terribly sorry to say this, but Cronicle v1 doesn't support HTTPS.  Several people have hacked it to sort of make it sort of work, but live log watching is a no-go, because it opens a secondary web socket directly to the worker server.

The good news is, this is all fixed in Cronicle v2, which is coming this year (2024).  Unfortunately it would be way too much work to backport full HTTPS support into v1.

**Reply by DerekF35** [reply]
@jhuckaby I apologize if this is answered elsewhere but I couldn't find in my digging.  Is `croncile-edge` the codebase for your Cronicle v2 release or a completely unrelated fork?

**Reply by jhuckaby** [reply]
@DerekF35 Totally unrelated fork.  I wish the author hadn't created a GitHub org named "cronicle-edge" because it makes it look very official, when it is not.

Don't get me wrong, he's an awesome dude, and very talented.  I just wish he left his fork under his personal GH account.

**Reply by DerekF35** [reply]
@jhuckaby Thanks for the quick reply and your answer makes sense.  If that was the future I was considering on jumping ahead, but I will wait patiently for your implementation.

**Reply by jhuckaby** [reply]
Well, don't let me dissuade you from trying it!  Mike added a ton of good features in there, including (I believe) HTTPS support, even for live log watching, Docker image, etc.

It's definitely worth a look 😊
